BreakingBad/CharacterListScreen.swift

```diff
@@ -56,46 +56,56 @@ class CharacterListScreen: UIViewController {
     
      
     private func createCharacterList(from characterList : [CharacterJson]) -> [Character] {
-
         var tempCharacters: [Character] = []
-
         for character in characterList {
-            //let imageURL = URL(string: character.img)
+            let image = defaultImage!
             let birthday: String = character.birthday
             let age = addAgeToBirthday(from: birthday)
-            let image = defaultImage! //getImage(from: imageURL)
             let newCharacter = Character(name: character.name, birthday: age, image: image, nickname: character.nickname, occupations: character.occupation, portrayed: character.portrayed)
             tempCharacters.append(newCharacter)
-            
+            if let imageURL = URL(string: character.img) {
+                getImageAsync(from: newCharacter, imageUrl: imageURL)
+            }
         }
         return tempCharacters
     }
     
     private func addAgeToBirthday(from birthday: String) -> String {
-        if birthday != "Unknown" {
-            
-            let dateFormatter = DateFormatter()
-            dateFormatter.dateFormat = "MM'-'dd'-'YYY'"
-            let date = dateFormatter.date(from: birthday)
-            let now = Date()
-            guard let formattedBirthday: Date = date else {
+            if birthday != "Unknown" {
+                
+                let dateFormatter = DateFormatter()
+                dateFormatter.dateFormat = "MM'-'dd'-'YYY'"
+                let date = dateFormatter.date(from: birthday)
+                let now = Date()
+                guard let formattedBirthday: Date = date else {
+                    return birthday
+                }
+                let calendar = Calendar.current
+                let ageComponents = calendar.dateComponents([.year], from: formattedBirthday, to: now)
+                let age = birthday + " (" + String(ageComponents.year!) + ")"
+                return age
+            }
+            else {
                 return birthday
             }
-            let calendar = Calendar.current
-            let ageComponents = calendar.dateComponents([.year], from: formattedBirthday, to: now)
-            let age = birthday + " (" + String(ageComponents.year!) + ")"
-            return age
         }
-        else {
-            return birthday
+    
+    private func getImageAsync(from character: Character, imageUrl: URL) {
```

Comment:
Once the closure on this method completes, it will set the image on your model, but then what? Your table view will have already finished drawing itself and will have loaded the default image. It has no knowledge that the model has been refreshed with new data. 

To get the new image to show you would need to either reload the whole table view or the cell that contains this image. See if you can figure out how to do that.